### ISSUE TITLE
Does Netmiko support user pri level 2-14?

### ISSUE THREAD

**Issue by networkjourney** [ktbyers/netmiko#2152]
Hey Kirk/Folks
Does netmiko support user priviledge level 2 to 14?

I can use .enable() with priv level 1 but not with other levels.

What is the turnaround?

Or it don't support at all.

**Comment by ktbyers** [comment]
What does your prompt look like in privilege level 2 - 14?

And what is the issue you are running into--just failing on elevating privileges via `enable()`?

Can you post the exception stack trace for a failing case.

**Comment by networkjourney** [comment]
Yes the prompt looks to be 
SW01#
for privilege 2-14.

Could you help with .read_channel() and .write_channel() in this situation.

**Comment by networkjourney** [comment]
Regards to priv 2 error,
===================
**MY CODE:**
`    net_connect = ConnectHandler(**ciscodevice123)
    net_connect.enable()
    output = net_connect.send_command("show run")
    #output = net_connect.enable()
    #config_cmds = ['interface gi 0/0', 'descr connect SVR1']
    #output = net_connect.send_config_set(config_cmds)
    print(output)`

**ERROR:**
`SSH connection established to 192.168.32.209:22
**Interactive SSH session established
           ^
% Invalid input detected at '^' marker.**

Process finished with exit code 0`

===============
**PROMT ON MY CISCO DEVICE FOR PRI LEVEL 2:**
SW06#

So basically the .enable() inside my script is not effective for privilege level 2-14
What could be turnaround.

**Comment by ktbyers** [comment]
Okay, you probably would need to make your own function to handle this. Basically the Cisco prompt is identical from privilege 2 - 14 when compared to 15 (so Netmiko will already think it has elevated privileges).

It would be something like:

```python
    output = ""
    net_connect.write_channel("enable\n"))
    time.sleep(1)
    output += net_connect.read_until_prompt_or_pattern(pattern=r"ssword")  
    net_connect.write_channel(f"{net_connect.secret}\n")
    time.sleep(1)
    output += net_connect.read_until_prompt()
```

You could throw in a `show privilege` at the end of it (if you wanted to verify it was successful).

**Comment by networkjourney** [comment]
Thank you , your codes works perfectly.
Do you mind explaining some of line here.
a. what is output = "" ?
b. can we not just use output = net_connect.read_channel(pattern="password") , what is r"ssword"
c. also what is f in f"{net_connect.secret\n}
d. why have you used output += ? what is += ??


====================

I tried simplifying your given codes and this worked as well

    net_connect.write_channel("enable\n")
    output1 = net_connect.read_until_prompt(pattern="Password")
    net_connect.write_channel(f"{net_connect.secret}\n")
    output = net_connect.send_command("show run")
    print(output)

**Comment by ktbyers** [comment]
Yeah, I didn't need that first: `output = ""`, I could have just done:

```python
    net_connect.write_channel("enable\n"))
    time.sleep(1)
    output = net_connect.read_until_prompt_or_pattern(pattern=r"ssword")  
    net_connect.write_channel(f"{net_connect.secret}\n")
    time.sleep(1)
    output += net_connect.read_until_prompt()
```

**Comment by ktbyers** [comment]
The f"some string" is a Python f-string. You should be able to search on that online.

output += "something"

is identical to doing:

output = output + "something"

i.e. it will add the new string to the existing `output` string.

If you just use `read_channel(pattern=r"ssword")` and not `read_until_prompt_or_pattern(pattern=r"sswrod"), then your code will fail in any situation that you call `enable()` but are already in `privilege 15`.

The `r` before the string isn't needed here, but it is used to escape special characters in Python so that they don't mess up the regular expression search (i.e. pattern is used in a regular expression search). But our string doesn't have any special characters so it is not necessary.

**Comment by networkjourney** [comment]
net_connect.write_channel("enable\n")
net_connect.write_channel(f"{net_connect.secret}\n")

one last question
Both stmnts above are to write on channgel
so why on one just "enable" and on second we are using f-strings?

**Comment by ktbyers** [comment]
Because in the second one you need to send in a variable (the f-strings make it easier to embed a variable in a string.
